natlas-server/app/api/processing/screenshot.py

```diff
@@ -2,54 +2,75 @@
 import base64
 import hashlib
 import os
-from PIL import Image
+from io import BytesIO
+from PIL import Image, UnidentifiedImageError
+
+
+def is_valid_image(path: str) -> bool:
+    try:
+        Image.open(path)
+        return True
+    except (FileNotFoundError, UnidentifiedImageError):
+        return False
+
+
+def get_file_ext(service: str) -> str:
```

Comment:
Let me know if your refactor will fix this.

This makes a big assumption about another block of code. If we ever change the screenshot format, we could forget about this. It would be better to have the screenshotter register a screenshot and it's format in the scan object.

Reply:
Yeah the agent refactor will change the way the data is stored in the scan so that it includes file extension with the current data, then we can get rid of this function. The only reason it's being delayed is to not cause backwards incompatibility or try to maintain a version compatibility layer.